src/pages/ProjectEditPage/hooks/useFileUpload.ts

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useRef, useState } from "react"
+
+import { postProjectFiles } from "@api/projectFile/postProjectFiles"
+
+import { FileUploadStateType } from "../types/FileUploadStateType"
+
+export const useFileUpload = () => {
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const [files, setFiles] = useState<
+    FileUploadStateType<{ fileUrl: string }>[]
+  >([])
+
+  const onChangeFile = async (e: Event, fileLength: number) => {
+    if (!(e.target as HTMLInputElement).files) return
+    const selectFiles = (e.target as HTMLInputElement).files as FileList
+    const uploadFiles = Array.from(selectFiles)
+
+    if (files.length + uploadFiles.length > 6) {
+      alert("더이상 추가할 수 없습니다")
+      return
+    }
+
+    const promises = Object.values(uploadFiles).map((file, idx) => {
+      const imageFormData = new FormData()
+      imageFormData.append("file", file)
+
+      const index = fileLength + idx - 1 < 0 ? 0 : fileLength + idx
+
+      setFiles((prev) => [
+        ...prev,
+        { isLoading: true, error: null, data: null },
+      ])
+
+      return postProjectFiles(imageFormData)
+        .then((data) => {
+          setFiles((prev) => {
+            console.log(index)
+
+            const updatedFiles = [...prev]
+            updatedFiles[index] = {
+              isLoading: false,
+              error: null,
+              data,
+            }
+            return updatedFiles
+          })
+        })
+        .catch((error) => {
+          // 실패 시 파일 상태 업데이트
+          setFiles((prev) => {
+            console.log(index)
+
+            const updatedFiles = [...prev]
+            updatedFiles[index] = {
+              isLoading: false,
+              error,
+              data: null,
+            }
+            return updatedFiles
+          })
```

Comment:
```suggestion
const updateFileState = (index, update) => {
    setFiles((prev) => {
      const updatedFiles = [...prev];
      updatedFiles[index] = { ...updatedFiles[index], ...update };
      return updatedFiles;
    });
};
```
[P3] : 요기는 위에거하고 동일한거 같아서 이런식으로 분리해도 좋을 것 같습니다~